File: spec/php/StrPadTermTest.php
<?php
namespace Kaitai\Struct\Tests;

class StrPadTermTest extends TestCase {
    public function testStrPadTerm() {
        $r = StrPadTerm::fromFile(self::SRC_DIR_PATH . "/str_pad_term.bin");

        $this->assertEquals('str1', $r->strPad);
        $this->assertEquals('str2foo', $r->strTerm);
        $this->assertEquals('str+++3bar+++', $r->strTermAndPad);
        $this->assertEquals('str4baz@', $r->strTermInclude);
    }
}

File: spec/go/str_pad_term_test.go
package spec

import (
	"os"
	"testing"

	"github.com/kaitai-io/kaitai_struct_go_runtime/kaitai"
	"github.com/stretchr/testify/assert"

	. "test_formats"
)

func TestStrPadTerm(t *testing.T) {
	f, err := os.Open("../../src/str_pad_term.bin")
	if err != nil {
		t.Fatal(err)
	}
	s := kaitai.NewStream(f)

	var r StrPadTerm
	err = r.Read(s, &r, &r)
	if err != nil {
		t.Fatal(err)
	}

	assert.EqualValues(t, r.StrPad, "str1")
	assert.EqualValues(t, r.StrTerm, "str2foo")
	assert.EqualValues(t, r.StrTermAndPad, "str+++3bar+++")
	assert.EqualValues(t, r.StrTermInclude, "str4baz@")
}
